opal/mca/common/ofi/configure.m4

```diff
@@ -18,15 +19,17 @@
 AC_DEFUN([MCA_opal_common_ofi_CONFIG],[
     AC_CONFIG_FILES([opal/mca/common/ofi/Makefile])
 
-    # Check for ofi.  Note that $opal_common_ofi_happy is
+    OPAL_CHECK_OFI
+
+    # Note that $opal_common_ofi_happy is
     # used in other configure.m4's to know if ofi configured
     # successfully.
-    OPAL_CHECK_OFI([opal_common_ofi],
-                   [opal_common_ofi_happy=yes
-                    common_ofi_WRAPPER_EXTRA_LDFLAGS=$opal_common_ofi_LDFLAGS
-                    common_ofi_WRAPPER_EXTRA_LIBS=$opal_common_ofi_LIBS
-                    $1],
-                   [opal_common_ofi_happy=no
-                    $2])
+    AS_IF([test "$opal_ofi_happy" = "yes"],
+          [opal_common_ofi_happy=yes
```

Comment:
Is this variable (`opal_common_ofi_happy`) used anywhere?

Reply:
Not yet but it supposed to be used in every component that has the common/ofi as a dependency.

Reply:
I'm thinking that the "revamp" commit intended `opal_ofi_happy` to be used, instead.  I.e., the One Variable to Rule Them All is the `opal_ofi_happy` -- which indicates that OFI configury itself was happy; it doesn't matter whether any individual OFI component's `happy` variable is set.

Reply:
I was thinking along the way that if I have a component that depends upon the common/ofi component then that component should check if common/ofi is happy. I know that this check is equivalent to checking opal_ofi_happy for now, but in the future the common component could depend on something new. It would be similar to what https://github.com/open-mpi/ompi/blob/d58c59eb71e4b116030522928ac65caa8a34ed7b/opal/mca/common/ucx/configure.m4#L17-L19 does. Though the naming scheme is different.

On the other hand https://github.com/open-mpi/ompi/blob/d58c59eb71e4b116030522928ac65caa8a34ed7b/opal/mca/common/cuda/configure.m4#L20 and https://github.com/open-mpi/ompi/blob/d58c59eb71e4b116030522928ac65caa8a34ed7b/opal/mca/common/sm/configure.m4#L26 don't seem to set similar happy values.

So I believe that in the future when some new dependency comes along, already having `opal_common_ofi_happy` might be useful. Especially for new people who are not accustomed to the code base, it is a good hint. Though I understand that more code, is more code that you need to maintain. So I could take it out if  you prefer.